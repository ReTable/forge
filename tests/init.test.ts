import { describe, expect } from 'vitest';

import { setup } from './setup';

const it = setup();

describe('init', async () => {
  await it(
    'generates initial config for browser',
    {
      command: 'init',
      name: 'browser-init-config',
      platform: 'browser',
    },
    async (c) => {
      expect(await c.read('.forgerc')).toMatchSnapshot();
    },
  );

  await it(
    'generates initial config for node',
    {
      command: 'init',
      name: 'node-init-config',
      platform: 'node',
    },
    async (c) => {
      expect(await c.read('.forgerc')).toMatchSnapshot();
    },
  );
});
