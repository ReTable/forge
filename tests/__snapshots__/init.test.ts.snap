// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`init > generates initial config for browser 1`] = `
"{
  \\"$schema\\": \\"https://github.com/ReTable/forge/blob/main/schemas/forgerc.json\\",
  \\"platform\\": \\"browser\\",
  \\"entry\\": \\"index\\",
  \\"check\\": true,
  \\"typings\\": true,
  \\"build\\": {
    \\"production\\": true
  },
  \\"watch\\": {
    \\"production\\": false,
    \\"storybook\\": true
  }
}"
`;

exports[`init > generates initial config for node 1`] = `
"{
  \\"$schema\\": \\"https://github.com/ReTable/forge/blob/main/schemas/forgerc.json\\",
  \\"platform\\": \\"node\\",
  \\"entry\\": \\"index\\",
  \\"check\\": true,
  \\"typings\\": true,
  \\"build\\": {
    \\"production\\": true
  },
  \\"watch\\": {
    \\"production\\": false
  }
}"
`;
